plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -307,6 +310,29 @@ public function add_woocommerce_remote_variant() {
 		}
 	}
 
+	/**
+	 * Set default option values for launch your store task.
+	 */
+	public function add_lys_default_values() {
+		$coming_soon      = current_action() === 'woocommerce_installed' ? 'yes' : 'no';
+		$store_pages_only = WCAdminHelper::is_site_fresh() ? 'yes' : 'no';
```

Comment:
I'm looking at the definition of `WCAdminHelper::is_site_fresh` and it may be problematic.

https://github.com/woocommerce/woocommerce/blob/e2ed2a9fdb08418d1c2c4881d8bc7a5d4ede8ff6/plugins/woocommerce/src/Admin/WCAdminHelper.php#L100-L108

1. It seems based on the current user
2. If the user account is less than a month old, there still may be live pages

Can we check if there is at least one record in `wp_posts` table with status set to `publish`? I'm wondering if the only safe thing to do here is always just make it store pages. Maybe this is worthy of further discussion amongst the team.

Reply:
Thanks for the early feedback! @psealock  🙂

 > Can we check if there is at least one record in wp_posts table with status set to publish?

I think it's safe to assume that if `is_site_fresh` returns true then users didn't publish any post or page yet because `WCAdminHelper::is_site_fresh` also checks the `fresh_site` option.

> I'm wondering if the only safe thing to do here is always just make it store pages. Maybe this is worthy of further discussion amongst the team.

That seems like a good idea. @adrianduffell What do you think?

Reply:
> It seems based on the current user

In hindsight I think we should change this to the first user record in the database. If we agree this is better, I'll create a follow-up issue.

> If the user account is less than a month old, there still may be live pages. Can we check if there is at least one record in wp_posts table with status set to publish?

The `fresh_site` option check should account for live pages -- it will have a value of `0` if content has been published. However it might still be worth making a `wp_posts` record check to give greater confidence. We would just have to ignore WooCommerce pages that get added automatically.

> I'm wondering if the only safe thing to do here is always just make it store pages.

There is a key project requirement is to apply "coming soon" to all pages when we're certain it's a fresh site. So perhaps I'd say we should just keep improving the `is_site_fresh` function until we're confidence about it.